src/coreclr/tools/aot/DependencyGraphViewer/Program.cs

```diff
@@ -143,16 +146,125 @@ struct GraphEvent
         public int Num3;
         public string Str;
     }
+    class DGMLGraphProcessing
+    {
+        ConcurrentQueue<GraphEvent> events = new ConcurrentQueue<GraphEvent>();
+
+        public delegate void OnCompleted(int currFileID);
+        public event OnCompleted Complete;
+
+        public DGMLGraphProcessing(int file)
+        {
+            FileID = file;
+        }
+        public int FileID
+        {
+            get; init;
+        }
+
+        public void FindXML(string argPath)
+        {
+            var fileContent = string.Empty;
+            var filePath = string.Empty;
+            var fileStream = System.IO.Stream.Null;
+
+            using (OpenFileDialog openFileDialog = new OpenFileDialog())
+            {
+                openFileDialog.InitialDirectory = argPath;
+                openFileDialog.Filter = @"All Files|*.*|DGML(*.dgml)|*.dgml|XML(*.xml)|*xml";
+                openFileDialog.FilterIndex = 2;
+                openFileDialog.RestoreDirectory = true;
+
+                if (!argPath.EndsWith(".dgml") && !argPath.EndsWith(".xml"))
+                {
+                    if (openFileDialog.ShowDialog() == DialogResult.OK)
+                    {
+                        //Get the path of specified file
+                        filePath = openFileDialog.FileName;
+                        fileStream = openFileDialog.OpenFile();
+                    }
+                }
+                else
+                {
+                    if (argPath.EndsWith(".dgml")) // all .dgml files in the path need to have .xml added for XmlReader to function
+                    {
+                        argPath += ".xml";
+                    }
+                    fileStream = new FileStream(argPath, FileMode.Open);
+                }
+
+            }
+            Thread th = new Thread(ParseXML);
+            if (fileStream != System.IO.Stream.Null)
+            {
+                th.Start(fileStream);
+            }
+
+        }
 
-    class GraphProcessing
+        private void ParseXML(object fileStream)
+        {
+            System.IO.FileStream fileContents = (System.IO.FileStream)fileStream;
+            GraphCollection collection = GraphCollection.Singleton;
+            XmlReaderSettings settings = new XmlReaderSettings();
+            settings.IgnoreWhitespace = true;
+
+            using (XmlReader reader = XmlReader.Create(fileContents, settings))
+            {
+                while (reader.Read())
+                {
+                    if (!(reader.Name.StartsWith("P"))) // skips over 'properties' in the .dgml file
+                    {
+                        IXmlLineInfo lineInfo = (IXmlLineInfo)reader;
+                        int lineNumber = lineInfo.LineNumber;
+                        switch (reader.NodeType)
+                        // fileCount is the PID for the system, increments down with the number of files being read.
```

Comment:
Good comment. I think it belongs on the `FileID` property.

Reply:
is the location of the comment where you would expect?